lib/app/router.dart

```diff
@@ -1,34 +1,50 @@
 import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:go_router/go_router.dart';
 import 'package:gruene_app/app/constants/routes.dart';
 import 'package:gruene_app/app/util/build_page_without_animation.dart';
 import 'package:gruene_app/app/widgets/main_layout.dart';
+import 'package:gruene_app/features/auth/bloc/auth_bloc.dart';
+import 'package:gruene_app/features/auth/bloc/auth_stream.dart';
+import 'package:gruene_app/features/auth/screens/login_screen.dart';
 import 'package:gruene_app/features/campaigns/screens/campaigns_screen.dart';
 import 'package:gruene_app/features/mfa/screens/mfa_screen.dart';
 import 'package:gruene_app/features/news/screens/news_screen.dart';
 import 'package:gruene_app/features/profiles/screens/profiles_screen.dart';
 import 'package:gruene_app/features/tools/screens/tools_screen.dart';
 
-GoRoute buildRoute(String path, Widget child) {
+GoRoute buildRoute(String path, Widget child, {bool withMainLayout = true}) {
   return GoRoute(
     path: path,
     pageBuilder: (BuildContext context, GoRouterState state) => buildPageWithoutAnimation<void>(
       context: context,
       state: state,
-      child: MainLayout(child: child),
+      child: withMainLayout ? MainLayout(child: child) : child,
     ),
   );
 }
 
-GoRouter createAppRouter() {
+GoRouter createAppRouter(AuthStream authStream) {
   return GoRouter(
     initialLocation: Routes.news,
+    refreshListenable: authStream,
```

Comment:
:question: Hmmm, I did not find any real documentation about except some people saying this might be deprecated. Could you please explain what this does/why this is needed?
Also stumbled upon this (perhaps interesting) comment on synchronization of bloc and go_router, does this make sense or is this already properly solved this way?

https://github.com/flutter/flutter/issues/116651#issuecomment-1853322247

Seriously, feel free to just say it does not make sense, I have neither worked with go_router nor with bloc, but perhaps it is still helpful.

Reply:
Thank your for this suggestion! I changed the code as suggested in the linked comment.